### PR TITLE
docs(button): add a story with all button options

### PR DESCRIPTION
**Pull request by jamime**
### Proposed behaviour

- Convert to CSF format
- Add new story with all button variations for visual regression testing

### Current behaviour

- Using deprecated`storiesOf` format
- Only posible to see each variation by changing the knobs

### Checklist

- [x] Release notes (Conventional Commits) 
<del>- [ ] Unit tests
<del>- [ ] Cypress automation tests
- [x] Storybook added or updated
<del>- [ ] Theme support
<del>- [ ] Typescript `d.ts` file added or updated

### Additional context
@sianford @aaron66 will use this to complete their applitools investigation. 

### Testing instructions

- `npm start`
- http://localhost:9001/iframe.html?id=button--all-buttons&theme=mint
- http://localhost:9001/iframe.html?id=button--all-buttons&theme=aegean
- http://localhost:9001/iframe.html?id=button--all-buttons&theme=none

_Note: these pages take a while to load due to the SVG icons._

### CODE REVIEW

File: src/components/button/button.stories.js [hunk 1]
```diff
@@ -38,45 +30,128 @@ const getKnobs = () => {
...
-    return (
-      <div>
-        <Button { ...props }>{ children }</Button>
-        <Button { ...props }>{ children }</Button>
-      </div>
-    );
-  };
+export const allButtons = () => {
+  const types = ['primary', 'secondary', 'tertiary'];
+  const sizes = ['small', 'medium', 'large'];
```
Comment (grabkowski) [inline src/components/button/button.stories.js:76]: you could use `OptionsHelper.sizesRestricted` here

File: src/components/button/button.stories.js [hunk 2]
```diff
@@ -38,45 +30,128 @@ const getKnobs = () => {
...
-    const { children } = props; // eslint-disable-line react/prop-types
-    return (
-      <div>
-        <Button { ...props }>{ children }</Button>
-        <Button { ...props }>{ children }</Button>
-      </div>
-    );
-  };
+export const allButtons = () => {
+  const types = ['primary', 'secondary', 'tertiary'];
```
Comment (grabkowski) [inline src/components/button/button.stories.js:75]: don't we also have a `darkBackground` type?

File: src/components/button/button.stories.js [hunk 3]
```diff
@@ -38,45 +30,128 @@ const getKnobs = () => {
...
+                </Button>
+
+                {size === 'large' && (
+                  <Button
+                    iconPosition={ iconPosition }
+                    iconType={ icon }
+                    size={ size }
+                    disabled
+                    buttonType={ type }
+                    subtext='line two'
```
Comment (grabkowski) [inline src/components/button/button.stories.js:146]: I think that you could extract `iconPosition, iconType, size and buttonType` to an object and just add these props as `{ ...objectName }`

File: src/components/button/button.stories.js [hunk 4]
```diff
@@ -38,45 +30,128 @@ const getKnobs = () => {
...
-        <Button { ...props }>{ children }</Button>
-        <Button { ...props }>{ children }</Button>
-      </div>
-    );
-  };
+export const allButtons = () => {
+  const types = ['primary', 'secondary', 'tertiary'];
+  const sizes = ['small', 'medium', 'large'];
+  return <>
+    {['before', 'after'].map(iconPosition => (
```
Comment (edleeks87) [inline src/components/button/button.stories.js:78]: Could use `buttonIconPositions` from OptionsHelper?

File: src/components/button/button.stories.js (hunk 2 above)
Comment (grabkowski) [reply src/components/button/button.stories.js:75]: if yes, then you could use `OptionsHelper.buttonTypes`

File: src/components/button/button.stories.js (hunk 3 above)
Comment (jamime) [reply src/components/button/button.stories.js:126]: I've done it in a new commit, can you let me know what you think. I'm not sure if it helps the readability of the code. I did the other OptionHelper too, but again I think it makes the code harder to read.

**Pull request merged**